File: Cpp/901_Online_Stock_Span.cpp
class StockSpanner {
public:
    StockSpanner() {
        
    }
    
    int next(int price) {
        int result = 1;
        for(int i=stock.size()-1; i>=0; --i){
            if(stock[i].first <= price) {
                result += stock[i].second;
                stock.pop_back();
            } else break;
        }
        stock.push_back(make_pair(price, result));
        
        return result;
    }
private:
    vector<pair<int, int>> stock;
};

/**
 * Your StockSpanner object will be instantiated and called as such:
 * StockSpanner* obj = new StockSpanner();
 * int param_1 = obj->next(price);
 */

File: Kotlin/901_Online_Stock_Span.kt
class StockSpanner() {
    
    val stock = Stack<Pair<Int, Int>>()
    
    fun next(price: Int): Int {
        var result = 1
        
        while(stock.isNotEmpty() && stock.peek().first <= price) {
                result += stock.pop().second
        }
        stock.push(Pair(price, result))
        
        return result
    }

}

/**
 * Your StockSpanner object will be instantiated and called as such:
 * var obj = StockSpanner()
 * var param_1 = obj.next(price)
 */